src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php

```diff
@@ -480,4 +483,54 @@ protected function getErrorMessageForException(Exception $e, array $messages)
             $exceptionCode
         );
     }
+
+    /**
+     * Process Grid search.
+     *
+     * @param Request $request
+     * @param string $gridDefinitionFactoryService
+     * @param string $redirectRoute
+     * @param array $redirectQueryParamsToKeep
+     *
+     * @return RedirectResponse
+     */
+    protected function redirectToFilteredGrid(
+        Request $request,
+        $gridDefinitionFactoryService,
+        $redirectRoute,
+        array $redirectQueryParamsToKeep = []
+    ) {
+        /** @var GridDefinitionFactoryInterface $definitionFactory */
+        $definitionFactory = $this->get($gridDefinitionFactoryService);
+        /** @var GridDefinitionInterface $definition */
+        $definition = $definitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+
+        $filtersForm = $gridFilterFormFactory->create($definition);
+        $filtersForm->handleRequest($request);
+
+        $redirectParams = [];
+        if ($filtersForm->isSubmitted()) {
+            if (empty($definition->getFiltersUuid())) {
+                $redirectParams = [
+                    'filters' => $filtersForm->getData(),
+                ];
+            } else {
+                $redirectParams = [
+                    $definition->getFiltersUuid() => [
+                        'filters' => $filtersForm->getData(),
+                    ],
+                ];
+            }
+        }
+
+        foreach ($redirectQueryParamsToKeep as $paramName) {
+            if ($request->query->has($paramName)) {
+                $redirectParams[$paramName] = $request->query->get($paramName);
+            }
+        }
+
+        return $this->redirectToRoute($redirectRoute, $redirectParams);
```

Comment:
I was thinking about different flow. 

Current filtering flow as I understand is something like this:
1. Click search in list
2. In controller's search action build grid's filters form & handle search request 
3. Redirect back to list with filters
4. In resolver, get filters from DB and then from request (weird behavior imo)
5. Persist resolved filters to DB 
6. Pass resolved filters to controller's action.

What if it was something like (key difference is bolded):
1. Click search in list
2. In controller's search action build grid's filters form, handle search request **& persist filters to DB**
3. Redirect back to list **without filters in query**
4. In resolver, get filters from DB (or use defaults) **(resolving from query params is removed)**
6. Pass resolved filters to controller's action.

wdyt?

Reply:
Step 1 is performed through a POST action. I have no idea why, I guess it's a legacy behavior, but this is important that we can perform filtering using GET filters. It's a very common practice in web applications. This is possible with current flow as step 3 is a redirect to a GET route. However in your solution, this cant be done anymore :/ .

As for the benefits of this standard practice, there's a lot: put filters in a browser favorite, pass an URL to a colleague so he can see the same page as you, use a link from another page that is already filtered ...

Reply:
I agree that this would be much easier and cleaner to save the filters in the controller rather than in the resolver. But the actual workflow has the advantage to allow really minimalist controllers.

Besides I agree that we should still be able to filter via get parameters. A compromise between these two behaviour could be to only save the filters in POST requests (filter controls action from the user) and not in GET request (but they still work). This way you can share a link with grid params but sill save the filters you selected manually.